### ISSUE TITLE
Issues within Code

### ISSUE THREAD

**Issue by godfearingman** [Entitynt/Roblox-Pattern-Scanner#1]
https://github.com/Entitynt/Roblox-Pattern-Scanner/blob/f728b4eea8ca65aadd69a7981b624e3da170f275/Roblox%20Pattern%20Scanner/NTDLL/NTDLL.cpp#L4-L11

These are useless imports, you're better off using the winapi functions instead of invoking the Nt ones.
1. This is a remote process, any calls to the reading or writing will have absolutely no benefit other than a few ns speed increase (which doesn't seem to be the reason of its use here)
2. Yes, you could argue that by calling the nt variants of unlocking memory you can pass a handle and lock memory outside of your process address space. However, after reading your code, there doesn't seem even to be a reason for you to unlock memory? It's never locked to begin with, nor do you even check if it is locked before doing any of that. 

I think you should re-look into what Nt(Un)lockvirtualmemory works, it's irrelevant in your code base and does not need to be used. It's undefined behaviour attempting to unlock memory that is not locked.

https://github.com/Entitynt/Roblox-Pattern-Scanner/blob/f728b4eea8ca65aadd69a7981b624e3da170f275/Roblox%20Pattern%20Scanner/Main.cpp#L16-L64

Please stick to one style, it's inconsistent to switch between tlhelp32 api and psapi. You can use tlhelp32 for enumerating process modules too, this'll remove the need for opening another process handle.

The constant global variables created in each header file for the wrongfully named global variable "Driver" could be made so much better, please follow the idiomaticity of C++ and define it safely if you need it to be a global variable. 

https://github.com/Entitynt/Roblox-Pattern-Scanner/blob/f728b4eea8ca65aadd69a7981b624e3da170f275/Roblox%20Pattern%20Scanner/Scanner/Scanner.cpp#L28-L74
In this use case, this is really inefficient, you're scanning for static functions that are already visible on the compiled binary. Just focus on scanning the section you know it's in. Yes, this is a general-purpose scanner but you should write specific code for this source.

https://github.com/Entitynt/Roblox-Pattern-Scanner/blob/f728b4eea8ca65aadd69a7981b624e3da170f275/Roblox%20Pattern%20Scanner/Main.cpp#L101
This is not a real way to represent hexadecimal values. It will represent the decimal version of the hex value.

Regardless, good job just a few trivial issues.

**Comment by godfearingman** [comment]
I understand you are new but I really hope this doesn't come across as rude, just to help you learn the language better.

**Comment by Entitynt** [comment]
> I understand you are new but I really hope this doesn't come across as rude, just to help you learn the language better.

your comment is so valuable to me that such comments are very useful for people like me who are just starting out. Yes, what you said is absolutely right, I developed it very quickly, it is not a project that I think a lot about it, and I understood my mistakes better with you. Thanks again for your comment, I learned new things, thanks to you, I will be more careful in my next projects.
